OAUTH_SETUP.md

```diff
@@ -0,0 +1,54 @@
+# GitHub OAuth Setup
```

Comment:
Move this file to the docs folder and also include more general documentation of the user functionality, logins, reset password, email links and how that works. Consider the existing documentation in the docs folder and whether it needs updating (database docs?).